### PR TITLE
[Refactor] Refactor CastExpr constructor to support Map and Struct

### PR DESCRIPTION
**Pull request by alvin-coding**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

### CODE REVIEW

**Comment by alvin-coding** [comment]
run starrocks_admit_test

File: be/src/exprs/cast_nested.cpp [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+#include "column/map_column.h"
+#include "column/struct_column.h"
+#include "exprs/cast_expr.h"
+#include "exprs/expr_context.h"
+
+namespace starrocks {
+
+StatusOr<ColumnPtr> CastMapExpr::evaluate_checked(ExprContext* context, Chunk* ptr) {
+    ASSIGN_OR_RETURN(ColumnPtr orig_column, _children[0]->evaluate_checked(context, ptr));
+    if (ColumnHelper::count_nulls(orig_column) == orig_column->size()) {
```
Comment (mofeiatwork) [inline be/src/exprs/cast_nested.cpp:25]: `!has_null()` ?

File: be/src/exprs/cast_expr.h [hunk 2]
```diff
@@ -96,4 +86,68 @@ class CastJsonToArray final : public Expr {
...
+    Expr* clone(ObjectPool* pool) const override { return pool->add(new CastArrayExpr(*this)); }
+
+private:
+    std::unique_ptr<Expr> _element_cast;
+};
+
+// cast one MAP to another M.
+// For example.
+//   cast MAP<tinyint, tinyint> to MAP<int, int> is OK
+// TODO(alvin): function is enough, but i
```
[the remark this answers is not in the document]
Comment (alvin-coding) [reply be/src/exprs/cast_expr.h]: Fixed

File: be/src/exprs/cast_nested.cpp [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+StatusOr<ColumnPtr> CastArrayExpr::evaluate_checked(ExprContext* context, Chunk* ptr) {
+    ASSIGN_OR_RETURN(ColumnPtr orig_column, _children[0]->evaluate_checked(context, ptr));
+    if (ColumnHelper::count_nulls(orig_column) == orig_column->size()) {
+        return ColumnHelper::create_const_null_column(orig_column->size());
+    }
+    // NOTE: const(nullable) case is handled by last if case
+    auto* array_column = down_cast<ArrayColumn*>(ColumnHelper::get_data_column(orig_column.get()));
+
+    ColumnPtr casted_element_column;
+    // cast key column
```
Comment (sduzh) [inline be/src/exprs/cast_nested.cpp:95]:
```suggestion
    // cast element column
```

**Review by sduzh — APPROVED** [review]

**Comment by sduzh** [comment]
run starrocks_be_unittest

**Review by imay — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**